numpy/_core/src/multiarray/scalartypes.c.src

```diff
@@ -705,7 +705,7 @@ format_@name@(@type@ val, npy_bool scientific,
     }
     else {
         return Dragon4_Positional_@Name@(&val,
-                        DigitMode_Unique, CutoffMode_TotalLength, precision,
+                        DigitMode_Exact, CutoffMode_TotalLength, precision,
```

Comment:
Scientific notation is just above this, so would be equally easy to change.

But... the testsuite shims currently added in this PR are just a small sample of the tradeoffs we'd have to consider, so I won't invest more time here until that's all considered and weighed/balanced.